test/support/utils.rb

```diff
@@ -12,13 +12,54 @@ def type(command)
     end
 
     def create_message fail_msg, test_info
-      "#{fail_msg} on #{test_info.mode} mode\n[DEBUGGER SESSION LOG]\n> #{test_info.backlog.join('> ')}#{debuggee_backlog test_info}"
-    end
+      debugger_msg = <<~DEBUGGER_MSG.chomp
+        ======================
+        ▼  Debugger Session  ▼
```

Comment:
could you remove `▼` Unicode characters for old terminals (like Command prompt on Windows)?

Reply:
ok I've updated the overall style too:

```
Failure: test_conditional_breakpoint_stops_if_condition_is_true(DEBUGGER__::BreakAtLinesTest):
  #<Test::Unit::AssertionFailedError:
  --------------------
  | Debugger Session |
  --------------------

  > [1, 10] in /var/folders/yg/hnbymwxd5pn7v94_clc59y6r0000gn/T/debugger20210830-31956-16eovnp.rb
  > =>   1| module Foo
  >      2|   class Bar
  >      3|     def self.a
  >      4|       "hello"
  >      5|     end
  >      6|
  >      7|     def b(n)
  >      8|       2.times do
  >      9|         n
  >     10|       end
  > =>#0        <main> at /var/folders/yg/hnbymwxd5pn7v94_clc59y6r0000gn/T/debugger20210830-31956-16eovnp.rb:1
  > break if: n == 1
  >
  > (rdbg:remote) break if: n == 1
#0  BP - Check  n == 1
  >
  > (rdbg:remote) continue


  --------------------
  | Debugger Session |
  --------------------
  --------------------
  | Debuggee Session |
  --------------------

  > EBUGGER: Session start (pid: 31972)
  > DEBUGGER: Debugger can attach via UNIX domain socket (/Users/st0012/.ruby-debug-sock/ruby-debug-st0012-31956)
  > DEBUGGER: wait for debugger connection...
  > DEBUGGER: Connected.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo:Module (NameError)
  > Error caused by  BP - Check  n == 1.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo::Bar:Class (NameError)
  > Error caused by  BP - Check  n == 1.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo::Bar:Class (NameError)
  > Error caused by  BP - Check  n == 1.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo:Module (NameError)
  > Error caused by  BP - Check  n == 1.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo::Baz:Module (NameError)
  > Error caused by  BP - Check  n == 1.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo:Module (NameError)
  > Error caused by  BP - Check  n == 1.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo::Bar:Class (NameError)
  > Error caused by  BP - Check  n == 1.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo:Module (NameError)
  > Error caused by  BP - Check  n == 1.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo:Module (NameError)
  > Error caused by  BP - Check  n == 1.


  --------------------
  | Debuggee Session |
  --------------------
  -------------------
  | Failure Message |
  -------------------

  TIMEOUT ERROR (10 sec) on UNIX DOMAIN SOCKET mode.>.
/Users/st0012/projects/debug/test/support/assertions.rb:66:in `assert_block'
/Users/st0012/projects/debug/test/support/utils.rb:105:in `rescue in debug_code'
/Users/st0012/projects/debug/test/support/utils.rb:93:in `debug_code'
test/debug/break_test.rb:471:in `test_conditional_breakpoint_stops_if_condition_is_true'
     468:     end
     469:
     470:     def test_conditional_breakpoint_stops_if_condition_is_true
  => 471:       debug_code(program) do
     472:         type 'break if: n == 1'
     473:         assert_line_text(/#0  BP - Check  n == 1/)
     474:         type 'continue'
```

Reply:
If we remove  `▼` Unicode characters, do we need to sandwich the logs between two blocks? I prefer the following logs.

```shell
Failure: test_conditional_breakpoint_stops_if_condition_is_true(DEBUGGER__::BreakAtLinesTest):
  #<Test::Unit::AssertionFailedError:
  --------------------
  | Debugger Session |
  --------------------

  > [1, 10] in /var/folders/yg/hnbymwxd5pn7v94_clc59y6r0000gn/T/debugger20210830-31956-16eovnp.rb
  > =>   1| module Foo
  >      2|   class Bar
  >      3|     def self.a
  >      4|       "hello"
  >      5|     end
  >      6|
  >      7|     def b(n)
  >      8|       2.times do
  >      9|         n
  >     10|       end
  > =>#0        <main> at /var/folders/yg/hnbymwxd5pn7v94_clc59y6r0000gn/T/debugger20210830-31956-16eovnp.rb:1
  > break if: n == 1
  >
  > (rdbg:remote) break if: n == 1
#0  BP - Check  n == 1
  >
  > (rdbg:remote) continue

  --------------------
  | Debuggee Session |
  --------------------

  > EBUGGER: Session start (pid: 31972)
  > DEBUGGER: Debugger can attach via UNIX domain socket (/Users/st0012/.ruby-debug-sock/ruby-debug-st0012-31956)
  > DEBUGGER: wait for debugger connection...
  > DEBUGGER: Connected.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo:Module (NameError)
  > Error caused by  BP - Check  n == 1.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo::Bar:Class (NameError)
  > Error caused by  BP - Check  n == 1.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo::Bar:Class (NameError)
  > Error caused by  BP - Check  n == 1.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo:Module (NameError)
  > Error caused by  BP - Check  n == 1.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo::Baz:Module (NameError)
  > Error caused by  BP - Check  n == 1.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo:Module (NameError)
  > Error caused by  BP - Check  n == 1.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo::Bar:Class (NameError)
  > Error caused by  BP - Check  n == 1.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo:Module (NameError)
  > Error caused by  BP - Check  n == 1.
  > [EVAL ERROR]
  >   expr: n == 1
  >   err: undefined local variable or method `n' for Foo:Module (NameError)
  > Error caused by  BP - Check  n == 1.

  -------------------
  | Failure Message |
  -------------------

  TIMEOUT ERROR (10 sec) on UNIX DOMAIN SOCKET mode.>.
/Users/st0012/projects/debug/test/support/assertions.rb:66:in `assert_block'
/Users/st0012/projects/debug/test/support/utils.rb:105:in `rescue in debug_code'
/Users/st0012/projects/debug/test/support/utils.rb:93:in `debug_code'
test/debug/break_test.rb:471:in `test_conditional_breakpoint_stops_if_condition_is_true'
     468:     end
     469:
     470:     def test_conditional_breakpoint_stops_if_condition_is_true
  => 471:       debug_code(program) do
     472:         type 'break if: n == 1'
     473:         assert_line_text(/#0  BP - Check  n == 1/)
     474:         type 'continue'

```

Reply:
I still prefer the sandwich style because it still clearly indicates what the above session is when scrolling up through a long log.

I know I can "assume" the section above `Failure Message` would be `Debuggee Session`. But I think it's more friendly to indicate that as well.

Reply:
I see... but I prefer no-sandwich.
The sandwich style is a bit too much in my opinion.

First, our problem is 
> The separation between debugger log and the failure message isn't clear. 

It's clear to identify respective logs. The problem is solved without sandwich style.